addon/mixins/scoped-query-cache.js

```diff
@@ -68,10 +73,38 @@ export default Ember.Mixin.create({
       assert('shouldCache needs to be a boolean value', typeof(shouldCache) === 'boolean');
 
       if (shouldCache) {
-        queriesForModel[cacheReadStringifiedQuery] = queryPromise;
+        this._scopedQueryCache.cache.add(modelName, cacheReadStringifiedQuery, queryPromise);
       }
 
       return results;
     });
+  },
+
+  scheduleCacheDecay(delay = CACHE_DECAY_TIMEOUT, callback) {
+    if (this._scopedQueryCache.decayId) {
+      this.cancelCacheDecay();
+    }
+
+    this._scopedQueryCache.decayId = run.later(() => {
```

Comment:
We should likely use `setTimeout` here, as a `run.later` will cause tests to pause/wait until completion. I suspect a cache decay should not be waited on in testing.